tests/unit/lifecycle/LifecycleQueuePopulator.spec.js

```diff
@@ -172,4 +173,162 @@ describe('LifecycleQueuePopulator', () => {
             });
         });
     });
+
+    describe(':filter', () => {
+        let lcqp;
+        beforeEach(() => {
+            lcqp = new LifecycleQueuePopulator(params);
+            lcqp.locationConfigs = Object.assign({}, coldLocationConfigs, locationConfigs);
+        });
+        it('it should call _handleDeleteOp on delete message', () => {
+            const handleDeleteStub = sinon.stub(lcqp, '_handleDeleteOp').returns();
+            lcqp.filter({
+                type: 'delete',
+                bucket: 'lc-queue-populator-test-bucket',
+                key: 'hosts\x0098500086134471999999RG001  0',
+                value: JSON.stringify(templateEntry),
+            });
+            assert(handleDeleteStub.calledOnce);
+        });
+    });
+
+
+    describe(':_handleDeleteOp', () => {
+        const kafkaSendStub = sinon.stub().yields();
+        const objMd = {
+            'md-model-version': 2,
+            'owner-display-name': 'Bart',
+            'owner-id': '79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be',
+            'x-amz-storage-class': 'dmf-v1',
+            'content-length': 542,
+            'content-type': 'text/plain',
+            'last-modified': '2017-07-13T02:44:25.519Z',
+            'content-md5': '01064f35c238bd2b785e34508c3d27f4',
+            'key': 'object',
+            'location': [],
+            'isDeleteMarker': false,
+            'isNull': false,
+            'archive': {
+                archiveInfo: {
+                    archiveId: '04425717-a65c-4e8a-95e1-fa1d902d9d9f',
+                    archiveVersion: 7504504064263669
+                },
+            },
+            'dataStoreName': 'dmf-v1',
+        };
+        let lcqp;
+        beforeEach(() => {
+            lcqp = new LifecycleQueuePopulator(params);
+            lcqp.locationConfigs = Object.assign({}, coldLocationConfigs, locationConfigs);
+            lcqp._producers[`${coldStorageGCTopicPrefix}us-east-1`] = {
+                send: kafkaSendStub,
+            };
+            lcqp._producers[`${coldStorageGCTopicPrefix}dmf-v1`] = {
+                send: kafkaSendStub,
+            };
+        });
+        afterEach(() => {
+            kafkaSendStub.reset();
+        });
+        [
```

Comment:
Should we add unit tests for when the location config is not found, and we process a delete?